### ISSUE TITLE
suboptimal_flops does not capture expressions using subtraction

### ISSUE THREAD

**Issue by shssoichiro** [rust-lang/rust-clippy#9526]
### Summary

If I have the following code:

```rust
12.0 * x - A
```

And I have the `suboptimal_flops` lint enabled, I expect to receive a warning with a suggestion to use the following code instead:

```rust
12.0f32.mul_add(x, -A)
```

Currently this code does not produce any warning.

### Lint Name

suboptimal_flops

### Version

```text
rustc 1.65.0-nightly (cf9ed0dd5 2022-09-15)
binary: rustc
commit-hash: cf9ed0dd5836201843d28bbad50abfbe1913af2a
commit-date: 2022-09-15
host: x86_64-unknown-linux-gnu
release: 1.65.0-nightly
LLVM version: 15.0.0
```
